tensorboard/webapp/metrics/store/metrics_types.ts

```diff
@@ -180,6 +179,19 @@ export interface MetricsNamespacedState {
    */
   unresolvedImportedPinnedCards: CardUniqueInfo[];
   cardMetadataMap: CardMetadataMap;
+  // Minimum and maximum step number across all TimeSeries data.
+  stepMinMax: {
+    min: number;
+    max: number;
+  };
+
+  /*
+   * Ui-related states: states below affects the dashboard appearance.
+   */
+  cardToPinnedCopy: CardToPinnedCard;
+  // A map of card ids that previously pinned by the user.
+  cardToPinnedCopyCache: CardToPinnedCard;
+  pinnedCardToOriginal: PinnedCardToCard;
   cardStateMap: CardStateMap;
```

Comment:
cardStateMap has both data-related (dataMinMax) and ui-related states (everything else).

Might be worth separating it out into a third category here and then similarly organizing it to clearly separate data-related and ui-related states.